### PR TITLE
added rail fence cipher and test

### PR DESCRIPTION
**Pull request by RituRajSingh878**
#### Brief description of what is fixed or changed
Updated and Added New RailFence cipher

#### Release Notes

- crypto
   - Added New Failfence cipher

### CODE REVIEW

**Comment by RituRajSingh878** [comment]
@asmeurer can you check it?

**Comment by oscargus** [comment]
This is exceptionally similar to #15534 including the typos in certain places. If you plan to continue working on it, makes sure you base it on #15534.

File: sympy/crypto/crypto.py [hunk 1]
```diff
@@ -2246,3 +2247,65 @@ def decipher_gm(message, key):
         m <<= 1
         m += not b
     return m
+
+########### RailFence Cipher #############
+
+def encipher_railfence(message,rails):
+    """
+    Performs Railfence Encryption on plaintext and returns ciphertext
```
Comment (czgdp1807) [inline sympy/crypto/crypto.py]: Add a blank line here.

File: sympy/crypto/crypto.py [hunk 2]
```diff
@@ -2246,3 +2247,65 @@ def decipher_gm(message, key):
         m <<= 1
         m += not b
     return m
+
+########### RailFence Cipher #############
+
+def encipher_railfence(message,rails):
+    """
```
Comment (czgdp1807) [inline sympy/crypto/crypto.py]: Add a blank line here.

File: sympy/crypto/crypto.py [hunk 3]
```diff
@@ -2246,3 +2247,65 @@ def decipher_gm(message, key):
...
+    Performs Railfence Encryption on plaintext and returns ciphertext
+    EXAMPLE
+    =======
```
Comment (czgdp1807) [inline sympy/crypto/crypto.py]: Add a blank line here.

File: sympy/crypto/crypto.py [hunk 4]
```diff
@@ -2246,3 +2247,65 @@ def decipher_gm(message, key):
...
+    >>> from sympy.crypto.crypto import encipher_railfence
+    >>> message = "hello world"
+    >>> encipher_railfence(message,3)
+    'horel ollwd'
+
+    Parameters
+    ==========
+
+    message(string) : the message to encrypt.
```
Comment (czgdp1807) [inline sympy/crypto/crypto.py]: Probably, `message: string, the message to encrypt`. Same for the below line.

File: sympy/crypto/crypto.py [hunk 5]
```diff
@@ -2246,3 +2247,65 @@ def decipher_gm(message, key):
...
+    rails(int) : the number of rails.
+
+    Returns
+    =======
```
Comment (czgdp1807) [inline sympy/crypto/crypto.py]: Add a blank line here.

File: sympy/crypto/crypto.py [hunk 6]
```diff
@@ -2246,3 +2247,65 @@ def decipher_gm(message, key):
...
+    The Encrypted string message.
+
+    References
+     ==========
```
Comment (czgdp1807) [inline sympy/crypto/crypto.py]: Formatting issue.

**Comment by czgdp1807** [comment]
@RituRajSingh878 This PR LGTM. Can you please resolve the conflicts and complete your work on this? Thanks.

**Comment by czgdp1807** [comment]
@RituRajSingh878 Please add `Closes https://github.com/sympy/sympy/pull/15534` in [this comment](https://github.com/sympy/sympy/pull/15789#issue-245107058).

**Comment by RituRajSingh878** [comment]
I am not free, so if you wish to work, you can.
Thanks

**Pull request closed without merging**